samples/cpp/benchmark_app/inputs_filling.cpp

```diff
@@ -93,6 +96,59 @@ ov::Tensor create_tensor_from_image(const std::vector<std::string>& files,
     return tensor;
 }
 
+template <typename T>
+ov::Tensor create_tensor_from_numpy(const std::vector<std::string>& files,
+                                    size_t inputId,
+                                    size_t batchSize,
+                                    const benchmark_app::InputInfo& inputInfo,
+                                    const std::string& inputName,
+                                    std::string* filenames_used = nullptr) {
+    size_t tensor_size =
+        std::accumulate(inputInfo.dataShape.begin(), inputInfo.dataShape.end(), 1, std::multiplies<size_t>());
+    auto allocator = std::make_shared<SharedTensorAllocator>(tensor_size * sizeof(T));
+    auto data = reinterpret_cast<T*>(allocator->get_buffer());
+
+    std::vector<std::shared_ptr<uint8_t>> numpy_array_pointers;
+    numpy_array_pointers.reserve(batchSize);
+
+    size_t numpy_batch_size = 1;
+    if (!inputInfo.layout.empty() && ov::layout::has_batch(inputInfo.layout)) {
+        numpy_batch_size = batchSize;
+    } else {
+        slog::warn << inputName
+                   << ": layout is not set or does not contain batch dimension. Assuming that numpy array "
+                      "contains data for all batches."
+                   << slog::endl;
+    }
+
+    for (size_t b = 0; b < numpy_batch_size; ++b) {
+        auto inputIndex = (inputId + b) % files.size();
+        if (filenames_used) {
+            *filenames_used += (filenames_used->empty() ? "" : ", ") + files[inputIndex];
+        }
+        FormatReader::ReaderPtr numpy_array_reader(files[inputIndex].c_str());
+        if (numpy_array_reader.get() == nullptr) {
+            slog::warn << "Numpy array " << files[inputIndex] << " cannot be read!" << slog::endl << slog::endl;
+            continue;
+        }
+
+        std::shared_ptr<uint8_t> numpy_array_data_pointer(
+            numpy_array_reader->getData(inputInfo.width(), inputInfo.height()));
```

Comment:
What if input is not image?

Reply:
Good point - I implemented a type casting with uchar buffer to fix it:
```
            size_t type_bytes_size = sizeof(T);
    unsigned char* bytes_buffer = new unsigned char[type_bytes_size];

    for (size_t batch_nr = 0; batch_nr < numpy_batch_size; ++batch_nr) {
        for (size_t input_tensor_nr = 0; input_tensor_nr < tensor_size; ++input_tensor_nr) {
            size_t offset = batch_nr * tensor_size + input_tensor_nr;
            for(size_t byte_nr; byte_nr < type_bytes_size; ++byte_nr){
                bytes_buffer[byte_nr] = numpy_array_pointers.at(batch_nr).get()[offset * type_bytes_size + byte_nr];
            }
            data[offset] = *((T*)bytes_buffer);
        }
    }
    delete[] bytes_buffer;
```